src/main/java/org/datadog/jenkins/plugins/datadog/traces/DatadogBaseBuildLogic.java

```diff
@@ -119,7 +119,11 @@ private List<StageData> traverseStages(List<FlowNode> heads, int limit) {
         Queue<FlowNode> nodes = new ArrayDeque<>(heads);
         while (!nodes.isEmpty()) {
             FlowNode node = nodes.poll();
-            nodes.addAll(node.getParents());
+            for (FlowNode parent : node.getParents()) {
+                if (parent != null){
```

Comment:
Wow, how a `FlowNode` parent can be null :O?

Reply:
This isn't a common situation for sure, maybe not even a "normal" one, as even the code in the Jenkins core starts throwing NPEs.